### PR TITLE
astyle

### PR DESCRIPTION
**Pull request by ftalbrecht**
This is not meant to be merged!

While clang-format clearly exceeds at many things it also lacks some functionality in other areas (see the other). In particular, I could not get clang-format to just leave some things alone.

So I started to experiment with other indenters to see what can be achieved. This is what can be done with astyle. There are also some things I do not like. The big difference to clang-format is, however, that astyle leaves many things alone, which might be good and bad.

### CODE REVIEW

File: dune/stuff/common/color.hh [hunk 1]
```diff
@@ -33,61 +33,59 @@ namespace Common {
...
-// light background colors
-static DS_CONST_CHAR bdarkgray;
-static DS_CONST_CHAR blightred;
-static DS_CONST_CHAR blightgreen;
-static DS_CONST_CHAR byellow;
-static DS_CONST_CHAR blightblue;
-static DS_CONST_CHAR blightpurple;
-static DS_CONST_CHAR blightcyan;
-static DS_CONST_CHAR bwhite;
+struct Colors {
```
Comment (ftalbrecht) [inline dune/stuff/common/color.hh]: I could not find a style to treat structs and enums like classes (to not attach the brackets).

File: dune/stuff/common/crtp.hh [hunk 2]
```diff
@@ -96,7 +96,7 @@ protected:
 
 #ifndef NDEBUG
 template <class I, class T>
-std::recursive_mutex CRTPInterface<I,T>::crtp_mutex_;
+std::recursive_mutex CRTPInterface<I, T>::crtp_mutex_;
```
Comment (ftalbrecht) [inline dune/stuff/common/crtp.hh]: There seems to be no option regarding the whitespace around angles.

File: dune/stuff/common/fixed_map.hh [hunk 3]
```diff
@@ -76,22 +82,29 @@ public:
 private:
   friend class boost::iterator_core_access;
 
-  void increment() { index_++; }
+  void increment()
```
Comment (ftalbrecht) [inline dune/stuff/common/fixed_map.hh]: I could not find an option to leave one-statement-blocks in one line. One could leave any blocks as one-liners if they already are, but that also includes blocks with an arbitrary amount of statements.

File: dune/stuff/functions/constant.hh [hunk 4]
```diff
@@ -108,7 +114,7 @@ class Constant
   : public GlobalFunctionInterface< EntityImp, DomainFieldImp, domainDim, RangeFieldImp, rangeDim, rangeDimCols>
 {
   typedef GlobalFunctionInterface< EntityImp, DomainFieldImp, domainDim, RangeFieldImp, rangeDim, rangeDimCols >
-      BaseType;
+  BaseType;
```
Comment (ftalbrecht) [inline dune/stuff/functions/constant.hh]: This is something that I do not like.

File: dune/stuff/functions/derived.hh [hunk 5]
```diff
@@ -141,18 +141,18 @@ public:
...
-                                   SelectDerived< FunctionType, derivative >::d,
-                                   typename SelectDerived< FunctionType, derivative >::R,
-                                   SelectDerived< FunctionType, derivative >::r,
-                                   SelectDerived< FunctionType, derivative >::rC >
+    typename SelectDerived< FunctionType, derivative >::D,
+    SelectDerived< FunctionType, derivative >::d,
+    typename SelectDerived< FunctionType, derivative >::R,
+    SelectDerived< FunctionType, derivative >::r,
+    SelectDerived< FunctionType, derivative >::rC >
 {
```
Comment (ftalbrecht) [inline dune/stuff/functions/derived.hh]: ouch

**Comment by ftalbrecht** [comment]
I left some comments in the code but did not go through all of the code. astyle is clearly inferior to clang-format in some areas, in particular when it comes to indentation associated with template arguments. On the other hand it leaves a lot of code as it is which is also nice since there is no reason to control everything. I will continue looking...

**Comment by renemilk** [comment]
Astyle has no understanding of the code it formats. Clang-format does. Imo you're wasting your time if you're looking for alternatives again.

**Comment by renemilk** [comment]
Imo branch can be deleted too

**Pull request closed without merging**